src/Service.Host/client/src/components/StorageTypes.js

```diff
@@ -0,0 +1,217 @@
+import React, { useEffect, useState } from 'react';
+import Typography from '@mui/material/Typography';
+import List from '@mui/material/List';
+import Grid from '@mui/material/Grid2';
+import { DataGrid } from '@mui/x-data-grid';
+import { Loading, ErrorCard, SnackbarMessage } from '@linn-it/linn-form-components-library';
+import Button from '@mui/material/Button';
+import Page from './Page';
+import config from '../config';
+import itemTypes from '../itemTypes';
+import usePut from '../hooks/usePut';
+import useInitialise from '../hooks/useInitialise';
+import usePost from '../hooks/usePost';
+import useGet from '../hooks/useGet';
+
+function StorageTypes() {
+    const { isLoading, result: storageTypesResult } = useInitialise(itemTypes.storageTypes.url);
+    const [creating, setCreating] = useState(false);
+    const [storageTypes, setStorageTypes] = useState();
+    const [snackbarVisible, setSnackbarVisible] = useState(false);
+    const [rowUpdated, setRowUpdated] = useState();
+
+    const {
+        send: getNewStorageTypes,
+        isNewStorageTypesLoading,
+        result: newStorageTypesGetResult
+    } = useGet(itemTypes.storageTypes.url);
+
+    const {
+        send: updateStorageType,
+        isLoading: updateLoading,
+        errorMessage: updateError,
+        putResult: updateResult,
+        clearPutResult: clearUpdateResult
+    } = usePut(itemTypes.storageTypes.url, true);
+
+    const {
+        send: createStorageType,
+        createStorageTypeLoading,
+        errorMessage: createStorageTypeError,
+        postResult: createStorageTypeResult,
+        clearPostResult: clearCreateStorageType
+    } = usePost(itemTypes.storageTypes.url);
+
+    useEffect(() => {
+        setStorageTypes(storageTypesResult);
+    }, [storageTypesResult]);
+
+    useEffect(() => {
+        if (!!updateResult || !!createStorageTypeResult) {
+            getNewStorageTypes();
+            setSnackbarVisible(true);
+            clearCreateStorageType();
+            clearUpdateResult();
+        }
+    }, [
+        clearCreateStorageType,
+        updateResult,
+        getNewStorageTypes,
+        createStorageTypeResult,
+        clearUpdateResult
+    ]);
+
+    useEffect(() => {
+        if (newStorageTypesGetResult) {
+            setStorageTypes(newStorageTypesGetResult);
+        } else {
+            setStorageTypes(storageTypesResult);
+        }
+    }, [newStorageTypesGetResult, storageTypesResult]);
```

Comment:
If you do delete useInitialise then you also won 't need this useEffect to swap between the two different lists